File: app/models/ticket.py
"""
Minimal Ticket model for MVP
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
from enum import Enum


class TicketStatus(str, Enum):
    OPEN = "open"
    RESOLVED = "resolved"


class Ticket(BaseModel):
    """
    Minimal Ticket Model
    No categories, priorities, or assignment.
    """
    user_id: str = Field(..., description="User ID who created the ticket")
    status: TicketStatus = Field(default=TicketStatus.OPEN)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    
    class Config:
        use_enum_values = True

File: models/Ticket.js
const mongoose = require('mongoose');

const TicketStatus = {
  OPEN: 'open',
  RESOLVED: 'resolved'
};

const ticketSchema = new mongoose.Schema({
  user_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  status: {
    type: String,
    enum: Object.values(TicketStatus),
    default: TicketStatus.OPEN,
    required: true
  }
}, {
  timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
});

ticketSchema.index({ user_id: 1 });
ticketSchema.index({ status: 1 });

const Ticket = mongoose.model('Ticket', ticketSchema);

module.exports = {
  Ticket,
  TicketStatus
};
